frontend/src/components/shared/ColorPicker/ColorPicker.test.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { shallow } from "lib/test_util"
+import { StatefulPopover as UIPopover } from "baseui/popover"
+import { ChromePicker } from "react-color"
+
+import ColorPicker, { Props } from "./ColorPicker"
+
+jest.mock("lib/WidgetStateManager")
+
+const getProps = (props: Partial<Props> = {}): Props => ({
+  label: "Label",
+  value: "#000000",
+  width: 0,
+  disabled: false,
+  onChange: jest.fn(),
+})
+const props = getProps()
+const wrapper = shallow(<ColorPicker {...props} />)
+const colorPickerWrapper = wrapper.find(UIPopover).renderProp("content")()
+
+describe("ColorPicker widget", () => {
+  it("renders without crashing", () => {
+    expect(wrapper.find(UIPopover).length).toBe(1)
+    expect(colorPickerWrapper.find(ChromePicker).length).toBe(1)
+  })
+
+  it("should render a label in the title", () => {
+    const wrappedDiv = wrapper.find("StyledColorPicker")
+    expect(wrappedDiv.find("StyledWidgetLabel").text()).toBe(props.label)
+  })
+
+  it("should have correct style", () => {
+    const wrappedDiv = wrapper.find("StyledColorPicker")
+    const { style } = wrappedDiv.props()
+
+    // @ts-ignore
+    expect(style.width).toBe(getProps().width)
+  })
+
+  it("should render a default color in the preview and the color picker", () => {
+    wrapper.find(UIPopover).simulate("click")
+    const chromePickerWrapper = wrapper.find(UIPopover).renderProp("content")()
+
+    expect(wrapper.find("StyledColorPreview").prop("style")).toEqual({
+      backgroundColor: "#000000",
+      boxShadow: "#000000 0px 0px 4px",
+    })
+
+    expect(chromePickerWrapper.prop("color")).toEqual("#000000")
+  })
+
+  it("supports hex shorthand", () => {
+    wrapper.find(UIPopover).simulate("click")
+
+    // this doesnt work either
```

Comment:
This is copied over from widgets color picker. Not sure what is not working 😅

Reply:
:thinking: probably worth just dropping the comment if it's unclear what's not working here